olm/operator-registry.md

```diff
@@ -0,0 +1,324 @@
+
+### Non-Goals
+
+  - Tools to build the index images or operator manifest images
+  - A definition for the operator manifest images or the content they contain
+
+## Proposal
+
+### User Stories
+
+_**Index Container Format**_
+
+As an operator author, I want to have a way of defining where my operator is released in an index of many operators:
+
+- So that I can publish my operator to a wide audience
+- So that I can define what my operator’s name and release information is
+
+Acceptance Criteria:
+
+A format is defined so that I can publish index container images to a container registry that are well defined and understood.
+
+**Internal details**:
+
+All the metadata that drives the on-cluster UI and install flow needs to be pulled by the cluster in order to display everything. Therefore it may be better to just build the index as a startable and queriable container that serves the content rather than a list of pointers to other content that needs to be pulled at runtime. There are some reasons to build the index as a simple list, largely stemming from a need for CI and build tools to build these indexes very rapidly. However, if the tools to generate the index are sufficiently well designed then bottlenecks can be avoided.
+
+The format then looks something like the operator-registry database that exists today, but with only minimal information about every single container. The db schema will have a layer of "latest" for each operator name for all the metadata that drives the OpenShift console (descriptions, icon references, etc.). Then all operator image pointers will aggregate just the metadata that drives install and upgrade (version, channels, dependencies, owned crds). To add additional updates to the database, the database just needs to be inserted/batch updated with changes rather than rebuilt from scratch.
+
+This database will be built by tooling that also needs to be created, but is not in scope of this enhancement.
+
+---------------------
+
+_**Serve Operator Database**_
+
+As an Operator developer, I want my Operator to be consumed and made available to clusters.
+
+- So that the metadata attached to my operator can be used to determine what is available to install on a cluster.
+
+Acceptance Criteria:
+I can create a registry database that hosts a history of my operator and other developer's operators. This database can be used by openshift clusters.
+
+---------------------
+
+_**Add Indexes to cluster**_
+
+As an OpenShift administrator, I want to have a way of adding an operator index to my cluster
+
+- So that I can make operators available to install onto my cluster
+
+Acceptance Criteria:
+When I add an operator index to my cluster, I am able to get information about the operators defined in that index.
+
+Internal detail note:
+Create an image type CatalogSource that, when added to a cluster with a reference to an index image, pulls the index manifest down and generates the operator metadata database for that operator.
```

Comment:
The intention here was to reuse that type specifically so that we do not need to create a new one. Since in both cases they result in a database serving the catalog there is no need to create a separate type since the implementation remains the same. In fact, adding the polling implementation for this enhancement should also make polling work for the current full size registry images.